scripts/Muon/GUI/ElementalAnalysis/LoadWidget/load_utils.py

```diff
@@ -68,28 +67,80 @@ def search_user_dirs(run):
     return files
 
 
-def group_by_detector(run, workspaces):
+# merge each detector workspace into one
+def merge_workspaces(run, workspaces):
     """ where workspaces is a tuple of form:
             (filepath, ws name)
     """
     d_string = "{}; Detector {}"
+    # detectors is a dictionary of {detector_name : [names_of_workspaces]}
     detectors = {d_string.format(run, x): [] for x in range(1, 5)}
+    # fill dictionary
     for workspace in workspaces:
         detector_number = get_detector_num_from_ws(workspace)
         detectors[d_string.format(run, detector_number)].append(workspace)
+    # initialise a group workspace
+    tmp = mantid.CreateSampleWorkspace()
+    overall_ws = mantid.GroupWorkspaces(tmp, OutputWorkspace=str(run))
+    # merge each workspace list in dectors into a single workspace
     for detector, workspace_list in iteritems(detectors):
-        mantid.GroupWorkspaces(workspace_list, OutputWorkspace=str(detector))
+        if workspace_list:
+            # sort workspace list
+            workspace_list = sorted(workspace_list)
```

Comment:
instead of using sort here and having the hard coded indices in elemental analysis, it would be better for us to manually compile the list here and to have the the indices stored here. We can set it up to ensure that the two are always consistent. 

If this is unclear, we can talk about it.